tests/test_2_login_with_correct_details.py

```diff
@@ -1,181 +1,49 @@
-
-    register_first(driver, user_data)
-
-    login_text_present = element_present(
-        driver,
-        By.XPATH,
-        '//div[@class="login-form"]/h2[contains(text(), "Login to your account")]',
-    )
-    assert login_text_present == True
-
-    # find name textbox and type person's name
-    name_textbox = driver.find_element(By.XPATH, '//input[@data-qa="login-email"]')
-    name_textbox.send_keys(user_data["email"])
-
-    # find email textbox and type person's email address
-    email_textbox = driver.find_element(By.XPATH, '//input[@data-qa="login-password"]')
-    email_textbox.send_keys(user_data["password"])
-
-    # find login button and click it
-    login_button = driver.find_element(By.XPATH, '//button[@data-qa="login-button"]')
-    login_button.click()
-
-    logged_in_status = driver.find_element(
-        By.XPATH, '//ul[@class="nav navbar-nav"]/li[last()]/a'
-    )
-    assert (
-        logged_in_status.text
-        == "Logged in as " + user_data["first_name"] + " " + user_data["last_name"]
-    )
-
-    # find delete account button and click it
-    delete_account_button = driver.find_element(
-        By.XPATH, '//a[@href="/delete_account"]'
-    )
-    delete_account_button.click()
-
-    # verify there's label saying "Account Deleted!"
-    account_deleted_text_present = element_present(
-        driver,
-        By.XPATH,
-        '//h2[@data-qa="account-deleted"]/b[contains(text(), "Account Deleted!")]',
-    )
-    assert account_deleted_text_present == True
-
-    # Press continue button
-    continue_button = driver.find_element(By.XPATH, '//a[@data-qa="continue-button"]')
-    continue_button.click()
+    homepage = HomePage(init_driver)
+    login_page = LoginPage(init_driver)
+    signup_page = SignUpPage(init_driver)
+    account_actions_page = AccountActionPage(init_driver)
+
+    homepage.go_to_website()
+    homepage.check_website_visible()
+    homepage.login()
+
+    login_page.check_new_user_label()
+    login_page.signup_enter_name(user_data["first_name"])
+    login_page.signup_enter_email(user_data["email"])
+    login_page.click_signup()
+
+    signup_page.select_title(user_data["title"])
+    signup_page.enter_name(f"{user_data['first_name']} {user_data['last_name']}")
+    signup_page.enter_password(user_data["password"])
+    signup_page.select_dob(
+        user_data["dob"][0], user_data["dob"][1], user_data["dob"][2]
+    )
+    signup_page.check_newsletter()
+    signup_page.check_special_offers()
+    signup_page.enter_first_name(user_data["first_name"])
+    signup_page.enter_last_name(user_data["last_name"])
+    signup_page.enter_address(user_data["address"])
+    signup_page.select_country(user_data["country"])
+    signup_page.enter_state(user_data["state"])
+    signup_page.enter_city(user_data["city"])
+    signup_page.enter_zipcode(user_data["zipcode"])
+    signup_page.enter_mobile_number(user_data["mobile_number"])
+    signup_page.register()
```

Comment:
you should not be registering user in a test case that is supposed to be testing login.
i recommend using pytest fixtures for preparing user before test case. 
```
# WebDriver setup
@pytest.fixture(scope="module")
def driver():
    driver = webdriver.Chrome()
    driver.implicitly_wait(10)
    yield driver
    driver.quit()

# Fixture for creating a user
@pytest.fixture(scope="module")
def create_user():
    # Logic to create a user in your system
    user = {"username": "testuser", "password": "testpass"}
    # Add logic to actually create the user
    yield user
    # Optionally, you can add logic to clean up if user creation fails

# Fixture for logging in a user
@pytest.fixture
def login_user(driver, create_user):
    driver.get("http://example.com/login")
    login_page = LoginPage(driver=driver)
    login_page.login(create_user["username"], create_user["password"])

# Fixture for deleting a user
@pytest.fixture(scope="module")
def delete_user(create_user):
    yield
    # Logic to delete the user after tests are completed
    # Use create_user['username'] or a unique identifier to delete the user

# Test function
def test_user_login(driver, login_user, delete_user):  #driver fixture runs first, then login, then test case is performed and after that delete_user.
    # Your test logic after user login
    # Example: Verify successful login
    assert "Dashboard" in driver.title
```